src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -54,3 +69,66 @@ def getDrainInfo(cls):
         Return drainStats class variable
         """
         return cls.drainStats
+
+    def updateAgentSpeedDrainConfig(self, thresholdsHit):
+        """
+        Takes a list of speed drain configuration keys and updates the agent configuration
+        """
+        if 'CondorPriority' in thresholdsHit:
+            if self.condorAPI.editCondorJobs(
+                    "JobStatus=?=1 && (CMS_JobType =?= \"Production\" || CMS_JobType =?= \"Processing\")",
+                    "JobPrio", "999999"):
+                logging.info("Condor job priority updated to 999999 for pending Production/Processing jobs.")
+                self.speedDrainConfig['CondorPriority']['Enabled'] = True
```

Comment:
Before setting it to True (and updating the config), we should check whether it's false, otherwise it will keep updating this same key over and over.
In addition to that, I think we need to define a `updateConfig = False` at the very first line of this method.

Reply:
When the thresholds are checked, line 126 of DrainStatusPoller.py should prevent any unnecessary updates for this key

`if not v['Enabled'] or k == 'CondorPriority':`

Reply:
And that's why CondorPriority is always added to the enableKeys var, provided it's under the threshold. Then the actual couchdb update is done just below L99

Reply:
Oh wait a sec... I was looking at EnableAllSites instead of CondorPriority.  You are right.  Let me add the logic.